### PR TITLE
Implemented Dropbox Chooser API

### PR DESCRIPTION
**Pull request by LucasNVarella**
One of the two APIs that need to be implemented. Currently does not save to firebase, and it needs some more tie-ins with the UI.

### CODE REVIEW

File: src/FileUpload.js [hunk 1]
```diff
@@ -37,10 +39,15 @@ class FileUpload extends Component {
         return (
             <div className={classes.root}>
 
-                <FormGroup row>
+                <FormGroup row id={this.props.uploadName + this.props.fileName}>
                     <Fab size="small" color="primary" aria-label="Add" className={classes.fab}>
                         <AddIcon/>
                     </Fab>
+                    <Button
```
Comment (matticoli) [inline src/FileUpload.js]: Use FAB onClick, no need for second button

File: src/MediaUpload.js [hunk 2]
```diff
@@ -38,6 +38,30 @@ const styles = theme => ({
 
 });
 
+const dbxImageOptions = {
```
Comment (matticoli) [inline src/MediaUpload.js]: Move into dropbox driver, these constants don't need to be at the View level

File: src/MediaUpload.js [hunk 3]
```diff
@@ -38,6 +38,30 @@ const styles = theme => ({
...
+    },
+    cancel: function() { },
+    linkType: "preview", // or "direct"
+    multiselect: false,
+    extensions: ['images'],
+    folderselect: false,
+    sizeLimit: 1024 * 1024 * 1024, // in bytes
+};
+
+const dbxAudioOptions = {
```
Comment (matticoli) [inline src/MediaUpload.js]: Move into dropbox driver, these constants don't need to be at the View level

File: src/MediaUpload.js [hunk 4]
```diff
@@ -38,6 +38,30 @@ const styles = theme => ({
...
+    success: function(files) {
+        alert("Here's the file link: " + files[0].link)
+    },
+    cancel: function() { },
```
Comment (matticoli) [inline src/MediaUpload.js]: Is this property required or can we just omit it if it's an empty function? Can you use a lambda here for consistency?

File: src/MediaUpload.js [hunk 5]
```diff
@@ -69,7 +93,15 @@ class MediaUpload extends Component {
         let fileIndex = 0;
         let fileUploads = this.state.filesList.map((isUploaded) => {
             fileIndex++;
-            return (<FileUpload key={fileIndex} fileName={fileIndex}/>);
+            let dbxOptions;
+            if (this.props.uploadName === "Images") {
```
Comment (matticoli) [inline src/MediaUpload.js]: Pass file type to dbx choose function and do this logic in the dropbox driver

File: src/dropbox.js [hunk 6]
```diff
@@ -16,9 +23,27 @@ const dbx = {
...
+                });
+              }
         }
     },
 
+    isDropboxReady() {
+        return !!window.Dropbox;
+    },
+
+    onChoose(options) {
```
Comment (matticoli) [inline src/dropbox.js]: Have onChoose take in file type and callback and resolve options in here

File: src/MediaUpload.js (hunk 5 above)
Comment (LucasNVarella) [reply src/MediaUpload.js]: done

File: src/dropbox.js (hunk 6 above)
Comment (LucasNVarella) [reply src/dropbox.js]: done

**Pull request merged**